mne/decoding/transformer.py

```diff
@@ -715,3 +715,168 @@ def transform(self, X):
+        for param in trans_bandwidth_params.keys():
+            if not isinstance(param, (int, float)) and (param is not
+                                                        None) and (param !=
+                                                                   'auto'):
+                raise ValueError("%s must be of type int, float or None, "
+                                 "got %s instead" % (trans_bandwidth_params[
+                                     param], type(param)))
+
+        if not isinstance(self.filter_length, (int, str)) and (param is not
+                                                               None):
+            raise ValueError("filter_length must be a int, string or None. "
+                             "Got %s instead." % type(self.filter_length))
+
+        if not isinstance(self.n_jobs, int) and self.n_jobs == 'cuda':
+            raise ValueError('n_jobs must be int or "cuda", got %s instead.'
+                             % type(self.n_jobs))
+
+        if self.method not in ('fft', 'iir'):
+            raise ValueError('method must be either "fft" or "iir" or None. '
+                             'Got %s instead.' % self.method)
+
+        if self.l_freq == 0:
+            self.l_freq = None
+        if self.h_freq is not None and self.h_freq > (self.sfreq / 2.):
+            self.h_freq = None
+        if self.l_freq is not None and not isinstance(self.l_freq, float):
+            self.l_freq = float(self.l_freq)
+        if self.h_freq is not None and not isinstance(self.h_freq, float):
+            self.h_freq = float(self.h_freq)
+
+    def fit(self, X, y=None):
+        """Does nothing. For scikit-learn compatibility purposes.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data.
+        y : array, shape (n_epochs,) | None
+            The label for each epoch.
+
+        Returns
+        -------
+        self : instance of Filterer
+            Returns the modified instance.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("X should be of type ndarray (got %s)."
+                             % type(X))
+
+        return self
+
+    def transform(self, X, y=None):
+        """Filters data along the third axis.
+
+        Parameters
+        ----------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data.
+        y : None | array, shape (n_epochs,)
+            The label for each epoch.
+            If None not used. Defaults to None.
+
+        Returns
+        -------
+        X : array, shape (n_epochs, n_channels, n_times)
+            The data after filtering.
+        """
+        if not isinstance(X, np.ndarray):
+            raise ValueError("epochs_data should be of type ndarray (got %s)."
+                             % type(X))
+
+        X = np.atleast_3d(X)
+        shape = X.shape
+        X = X.reshape(shape[0] * shape[1], shape[2])
```

Comment:
+1

And change the docstring so that the filter applies to the last dimension whatever the dimensionality?